SDC/seed.js

```diff
@@ -0,0 +1,74 @@
+const faker = require( 'faker' );
+// const database = require( './index.js' );
+const Review = require( './Review.js' );
+const fs = require( 'fs' );
+const path = require( 'path' );
+const os = require( 'os' );
+const filename = path.join( __dirname, 'seedable.csv' );
+const writer = fs.createWriteStream( filename );
+writer.write('product_id,user,score,title,body,recommend,date,response_id,ease,value,quality,appearance,works');
+
+const generateScore = () => Math.ceil( Math.random() * 5 );
+
+const generateReviewScores = () => {
+  const subset = [ 'ease', 'value', 'quality', 'appearance', 'works' ];
+  const subsetScores = {
+    ease: generateScore(),
+    value: generateScore(),
+    quality: generateScore(),
+    appearance: generateScore(),
+    works: generateScore(),
+  };
+
+  if (Math.random() <= 0.75) {
+    const index = Math.ceil(Math.random() * subset.length);
+    const subsetName = subset[index];
+    subsetScores[subsetName] = 0;
+  }
+  return subsetScores;
+};
+
+const generateSeedReviewData = () => {
+  const data = [];
+  for (let i = 0; i < 10000; i += 1) { // creates chunks of data to be pushed to CSV
+    // randomly generate between 15 and 25 reviews per product
+    const numberOfReviews = Math.floor(Math.random() * 10) + 15;
+
+    for (let j = 0; j < numberOfReviews; j += 1) {
+      let review = faker.lorem.paragraph();
+      if (review.length > 255) {
+        review = review.slice(0, 255);
+      }
+      const subsetScores = generateReviewScores();
+      const params = {
+        product_id: i + 1,
+        user: faker.internet.userName(),
+        score: generateScore(),
+        title: faker.lorem.words(),
+        body: review,
+        recommend: Math.random() >= 0.5,
+        date: faker.date.past(),
+        response_id: j === 0 ? i : null,
+        ease: subsetScores.ease,
+        value: subsetScores.value,
+        quality: subsetScores.quality,
+        appearance: subsetScores.appearance,
+        works: subsetScores.works,
+      };
+      data.push(params);
+    }
+  }
+  return data;
+};
+for ( let n = 0; n < 5; n++ ) { // pushes chunks of data n times
+  let randomDat = generateSeedReviewData()
```

Comment:
So at this invocation of `generateSeedReviewData()`, you could pass in your current `n` for the loop, like `generateSeedReviewData(n)`. Then within that function, `product_id` could be: `(10,000 * n) + i` so that you wouldn't be overlapping on each iteration! You could swap that 10,000 out with however many entries you want to generate at a time, too.